CMG updates to Heppy

gpetruc opened the pull request:
Updates to Heppy that have already been integrated from CMG side
 * make sure jets are consistently sorted by pT even when a new set of jet energy corrections or a smearing are applied: https://github.com/CERN-PH-CMG/cmg-cmssw/pull/208
 * fix the met without muons: https://github.com/CERN-PH-CMG/cmg-cmssw/pull/218
 * fixed quark-gluon discriminator: https://github.com/CERN-PH-CMG/cmg-cmssw/pull/228
 * adding raw MET: https://github.com/CERN-PH-CMG/cmg-cmssw/pull/229
 * add more command line options to heppy binary: https://github.com/gpetruc/cmg-cmssw/commit/3e0c5754a7fea8fa03a27c208f8d39d6e431e270
 * tau POG updates: https://github.com/CERN-PH-CMG/cmg-cmssw/pull/241
 * photon isolation and id updates: https://github.com/CERN-PH-CMG/cmg-cmssw/pull/234 https://github.com/CERN-PH-CMG/cmg-cmssw/pull/227
 * jet cleaning from photons and raw MET https://github.com/CERN-PH-CMG/cmg-cmssw/pull/245 https://github.com/CERN-PH-CMG/cmg-cmssw/pull/249
 * c++ tool for lepton isolation recalculation ( 2303f3c33ccd69ee90bfa8b8fb900524072f70b7 )
 * mini isolation ( 6d0eabeb402631af436f28c74185cdeed5f89709 https://indico.cern.ch/event/368826/ )
 * more flexible jet-lepton cleaning: it can discard the lepton instead of the jet under if configured to do so, and saves the output of the discarded jets and leptons (cherry-picked from https://github.com/CERN-PH-CMG/cmg-cmssw/pull/223 )
 * faster isolated track veto using the c++ class above (https://github.com/CERN-PH-CMG/cmg-cmssw/pull/237 rebased) 
 * a few small updates to heppy (small speedup in some modules, better printout of `NTupleVariable` and similar objects, better timereport)

@arizzi @mariadalfonso

gpetruc commented:
Hi,

Any chance of getting this merged soon?

There's more to come from our side, in addition to #19, as you can see from the cmgtools github 
( https://github.com/CERN-PH-CMG/cmg-cmssw/pulls?q=is%3Apr+label%3A%22to+propagate+to+Colin%22+ ).
Should I add them here, rather than waiting for this to be done to make a new one?

_Giovanni_

cbernet commented:
Hi, 

You can add to this PR. 

C

Le 16 févr. 2015 à 14:29, Giovanni Petrucciani notifications@github.com a écrit :

> Hi,
> 
> Any chance of getting this merged soon?
> 
> There's more to come from our side, in addition to #19, as you can see from the cmgtools github 
> ( https://github.com/CERN-PH-CMG/cmg-cmssw/pulls?q=is%3Apr+label%3A%22to+propagate+to+Colin%22+ ).
> Should I add them here, rather than waiting for this to be done to make a new one?
> 
> Giovanni
> 
> —
> Reply to this email directly or view it on GitHub.

gpetruc commented:
Ok, I have added everything that exists now in the heppy branch in the CMGTools repository, and updated the description of the pull request to reflect this.
Note that the e/gamma MVA electron id is still left outside in #19
